shawerma/tests.py

```diff
@@ -0,0 +1,108 @@
+        self.assertEqual (response.status_code, 201)
+        client.logout()
+
+    def test_get_user_order(self):
+
+        client=APIClient()
+        client.login(username="Ali", password="password123")
+        query1 = Order.objects.filter(user=2)
+        response = client.get('/shawerma-history/2/')
+        q=12
+        self.assertEqual(response.status_code , 200)
+
+        client.logout()
+
+    def test_retrive_menu_item(self):
+        client = APIClient()
+        client.login(username="Ali", password="password123")
+        query1= MenuItem.objects.all()
+        response = client.get('/shawerma/')
+        self.assertEqual(response.status_code,200)
+        client.logout()
+
+    def test_best_customer(self):
+        client = APIClient()
+        client.login(username="Ali", password="password123")
+        queryset = Order.objects.filter(created_date__year = 2017).annotate(total=Sum(F('item__quantity') * F('item__menu__price'))).values('user_id', 'total').annotate(Count('user_id')).order_by('-total').values('user_id')[0]
+        response = client.get('/shawerma-best-customer/2017/')
+        self.assertEqual(response.data['user_id'],queryset['user_id'])
+        client.logout()
+
+    def test_avg_customer(self):
+        client = APIClient()
+        client.login(username="admin",password="password123")
+        queryset = Order.objects.filter(user=2).annotate(total=Avg(F('item__quantity') * F('item__menu__price'))).aggregate(avg1=Avg('total')).values()
+        response = client.get('/shawerma-createAVG-customer/2/')
+        self.assertEqual(response.data[0],queryset[0])
+        client.logout()
+
+    def test_monthly_report_admin(self):
+        client = APIClient()
+        client.login(username="admin", password="password123")
+        queryset = Order.objects.filter(created_date__year=2017).annotate(month=TruncMonth('created_date')).values('month').annotate(total=Sum(F('item__quantity') * F('item__menu__price'))).values('month', 'total')
+        response = client.get('/shawerma-monthly-report/2017/')
+        self.assertEqual(response.data[0],queryset[0])
+        client.logout()
+        #it is run
+
+    def test_monthly_report(self):
+        client = APIClient()
+        client.login(username="Ali", password="password123")
+        queryset = Order.objects.filter(created_date__year=2017).annotate(month=TruncMonth('created_date')).values('month').annotate(total=Sum(F('item__quantity') * F('item__menu__price'))).values('month', 'total')
+        response = client.get('/shawerma-monthly-report/2017/')
+        self.assertEqual(response.data,403)
+        client.logout()
+        #it is run
+
+    def test_log_in(self):
+        client = APIClient()
+       # user = User.objects.get(username='Ali')
+       # client.force_authenticate(user=user)
+        response = client.post('/shawerma_order/create/',{'address':'Ram'},format='json')
+        self.assertEqual (response.status_code,403)
```

Comment:
this should be `401` not `403`
`401` mean not logged in
`403` mean unauthorized - or logged in but don't have permission